autopatches/autopatch.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python3
+import json
+import sys
+import requests
+import os
+import subprocess
+
+catalog = requests.get("https://raw.githubusercontent.com/YunoHost/apps/master/apps.json").json()
+
+my_env = os.environ.copy()
+my_env["GIT_TERMINAL_PROMPT"] = "0"
+os.makedirs(".apps_cache", exist_ok=True)
+
+login = open("login").read().strip()
+token = open("token").read().strip()
+github_api = "https://api.github.com"
+
+
+def apps(min_level=4):
+
+    for app, infos in catalog.items():
+        if infos.get("state") == "working" and infos.get("level", -1) > min_level:
+            infos["id"] = app
+            yield infos
+
+
+def app_cache_folder(app):
+    return os.path.join(".apps_cache", app)
+
+
+def git(cmd, in_folder=None):
+
+    if not isinstance(cmd, list):
+        cmd = cmd.split()
+    if in_folder:
+        cmd = ["-C", in_folder] + cmd
+    cmd = ["git"] + cmd
+    return subprocess.check_output(cmd, env=my_env).strip().decode("utf-8")
+
+
+# Progress bar helper, stolen from https://stackoverflow.com/a/34482761
+def progressbar(it, prefix="", size=60, file=sys.stdout):
```

Comment:
https://github.com/tqdm/tqdm is easy to use for this.

Reply:
Indeed that's interesting, though one advantage of having this function in the code is that we don't have to pip install anything :s